File: Boost_Echo_Client/src/AckFollow_UnfollowMessage.cpp
//
// Created by MAY on 05/01/2022.
//

#include <iostream>
#include "../include/AckFollow_UnfollowMessage.h"

AckFollow_UnfollowMessage::AckFollow_UnfollowMessage(std::string _username, short _messageOpcode, short _opCode): AckMessage(_messageOpcode, _opCode), username(_username) {

}

std::string AckFollow_UnfollowMessage::toString() {
    std::string answer = "ACK ";
    answer.append(std::to_string(getMessageOpCode()));
    answer.append(" ");
    answer.append(username);
    return answer;
}

File: spl-net/src/main/java/bgu/spl/net/impl/messages/AckFollow_UnfollowMessage.java
package bgu.spl.net.impl.messages;

public class AckFollow_UnfollowMessage extends AckMessage{

    private String username;

    public AckFollow_UnfollowMessage(String username){
        super((short)4);
        this.username=username;
    }
    @Override
    public String toString(){
        short opcode=getOpCode();
        String output=String.valueOf(opcode);
        short messageOpcode=getMessageOpcode();
        output+=messageOpcode;
        output+=(username+'\0');
        return output;
    }

}
